### PR TITLE
[ENH] New plot for significance relationships

### PR DESCRIPTION
**Pull request by TonyBagnall**
Plotting tool under development to allow for the case where cliques can be deceiving. More to follow, PR so I dont forget where it is!

Edited David:
- Include new plot, that may be improved if any new idea is proposed. 
- Move statistical methods to performance_metrics module. It is used by CDDs and new significance plots. 
- Include more tests for stats (in performance_metrics module) and for CDDs and Significance plots (in visualisation module). 

Changes for a future PR: 
There is a common block of code for CDDs and Significance plots, the one that extracts cliques and/or p_values. 
1) The possibility of returning the p_values: as we give the option of using Nemenyi, which doesn't return p_values, and Wilcoxon, which does. This can be solved by raising an error (already in the CDD function).
2) I've thought in creating a new function named "get_cliques(test, ...), but is not easy to be designed as the way for building cliques differ from CDDs and Significance plots. CDD stops forming a clique when a gap is found, i.e. a method with significant differences, whereas Significance plots does not. For this I have to give a second thought.

### CODE REVIEW

**Comment by TonyBagnall** [comment]
will wait for #1002 for this

**Comment by MatthewMiddlehurst** [comment]
Good stuff moving this forward, think its good to try out at least. May conflict with #1092 but probably nothing major.

**Comment by dguijo** [comment]
Some examples of the new plot:

For the bake off redux. In this case, there is no difference as no gaps are present between classifiers. 
CD_redux
significance_redux

Including more classifiers shows the potential of this new plot:
CD_redux_complete
significance_redux_complete

More examples with TSER this time:
CD_TSER
significance_TSER

A more comprehensive comparison of TSER approaches:
CD_TSER_complete
significance_TSER_complete

**Review by hadifawaz1999 — APPROVED** [review]
LGTM

File: aeon/performance_metrics/tests/test_stats.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+"""Tests for the critical difference diagram maker."""
```
Comment (MatthewMiddlehurst) [inline aeon/performance_metrics/tests/test_stats.py:1]: This comment seems incorrect

File: aeon/performance_metrics/stats.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    """
+    Perform Wilcoxon test.
+
+    Parameters
+    ----------
+    results: np.array
+      results of estimators on datasets
+
+    lower_better : bool, default = False
+        Indicates whether smaller is better for the results in scores. For example,
```
Comment (MatthewMiddlehurst) [inline aeon/performance_metrics/stats.py:99]: extra space here

File: aeon/visualisation/results/_significance.py [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+    Returns
+    -------
+        fig : matplotlib.figure
+            Figure created.
+        ax : matplotlib.axes
+            Axes of the figure.
+        p_values : np.ndarray (optional)
+            if return_p_values is True, returns a (n_estimators, n_estimators) matrix of
+            unadjusted p values for the pairwise Wilcoxon sign rank test.
```
Comment (MatthewMiddlehurst) [inline aeon/visualisation/results/_significance.py:61]: Indented

**Review by MatthewMiddlehurst — COMMENTED** [review]
Looks good. Some comments that may be useful for another PR

**Pull request merged**